File: sandbox.client/front/ts/model/ClientDetails.ts
import {GenderType} from "./GenderType";
import {ClientPhone} from "./ClientPhone";
import {ClientAddr} from "./ClientAddr";
import {Charm} from "./Charm";
import {AddrType} from "./AddrType";

export class ClientDetails {
    public id: number/*int*/|null;
    public firstname: string;
    public lastname: string;
    public patronymic: string;
    public gender: GenderType;
    public dateOfBirth: string;
    public characterId: number;
    public addressOfResidence: ClientAddr;
    public addressOfRegistration: ClientAddr;
    public phone: ClientPhone[];

    constructor() {
        this.firstname = '';
        this.lastname = '';
        this.patronymic = '';
        this.gender = GenderType.MALE;
        this.dateOfBirth = '2018-08-22';
        this.characterId = new Charm().id;
        this.addressOfRegistration = new ClientAddr();
        this.addressOfRegistration.type = AddrType.REG;
        this.addressOfResidence = new ClientAddr();
        this.addressOfResidence.type = AddrType.FACT;
        this.phone = [new ClientPhone()];
    }

    public assign(o: any): ClientDetails {
        this.id = o.id;
        this.firstname = o.firstname;
        this.lastname = o.lastname;
        this.patronymic = o.patronymic;
        this.gender = o.gender;
        this.dateOfBirth = o.dateOfBirth;
        this.characterId = o.characterId
        this.addressOfRegistration.street = o.addressOfRegistration.street;
        this.addressOfRegistration.house = o.addressOfRegistration.house;
        this.addressOfRegistration.flat = o.addressOfRegistration.flat;
        this.addressOfRegistration.type = AddrType.REG;
        this.addressOfResidence.street = o.addressOfRegistration.street;
        this.addressOfResidence.house = o.addressOfRegistration.house;
        this.addressOfResidence.flat = o.addressOfRegistration.flat;
        this.addressOfResidence.type = AddrType.FACT;
        this.phone = o.phone;
        return this;
    }
}

File: sandbox.controller/src/kz/greetgo/sandbox/controller/model/model/ClientDetails.java
package kz.greetgo.sandbox.controller.model.model;


import java.util.ArrayList;
import java.util.Date;
import java.util.List;

public class ClientDetails {
    public Integer id;
    public String firstname;
    public String lastname;
    public String patronymic;
    public GenderType gender;

    public Date dateOfBirth;
//    public String dateOfBirth;
    public int characterId;
    public ClientAddr addressOfResidence;
    public ClientAddr addressOfRegistration;
    public List<ClientPhone> phone=new ArrayList<>();

    @Override
    public String toString() {

        return "ClientDetails{" +
          "id=" + id +
          ", firstname='" + firstname + '\'' +
          ", lastname='" + lastname + '\'' +
          ", patronymic='" + patronymic + '\'' +
          ", gender=" + gender +
          ", dateOfBirth=" + dateOfBirth +
          ", characterId=" + characterId +
          ", addressOfResidence=" + addressOfResidence +
          ", addressOfRegistration=" + addressOfRegistration +
          ", phone=" + phone +
          '}';
    }


//    public ClientDetails() {
//        this.firstname = "";
//        this.lastname = "";
//        this.patronymic = "";
//        this.gender = GenderType.MALE;
//        this.dateOfBirth = "2018-08-22";
//        this.characterId = new Charm().id;
//        this.addressOfRegistration = new ClientAddr();
//        this.addressOfRegistration.type = AddrType.REG;
//        this.addressOfResidence = new ClientAddr();
//        this.addressOfResidence.type = AddrType.FACT;
//        this.phone = new ArrayList<>();
//    }
}
